src/Components/Cart.js

```diff
@@ -0,0 +1,103 @@
+      }));
+      cartTotal = ownCart
+        .map(item => item.product.price * 1)
+        .reduce((acc, curr) => acc + curr, 0)
+        .toFixed(2);
+    } else {
+      userCart = [...guestCart];
+      cartTotal = userCart.map( item => item.price * 1 ).reduce((acc, curr) => acc + curr, 0).toFixed(2);
+    }
+
+    if (userCart.length === 0) {
+      return (
+        <div>
+          <h2>Your Cart</h2>
+          <div>Your cart is empty</div>
+        </div>
+      );
+    }
+    return (
+      <div className='container'>
+        <div className='cart'>
+          <h2>Your Cart</h2>
+          <ul className='collection'>
+            {
+              auth.name !== 'Guest' ?
+              ownCart.map(item => (
+              <li className ='collection-item avatar' key={item.id}>
+                <div className='item-img'>
+                  <img src={item.product.imageURL}/>
+                </div>
+                <div className='item-desc'>
+                  <span className="title">{item.product.name}</span>
+                  <p><b>Price: {item.product.price}$</b></p>
+                  <p>
+                      <b>Quantity:</b>
+                  </p>
+                </div>
+                  <div className='add-remove'>
+                  <i className="material-icons">arrow_drop_up</i>
+                  <i className="material-icons">arrow_drop_down</i>
+                  </div>
+                <button className='waves-effect waves-light btn orange darken-3 remove'onClick={() => deleteGuestItem(item)}>Remove Item</button>
+              </li>
+            )) : userCart.map(item => (
+              <li className ='collection-item avatar' key={item.id}>
+                <div className='item-img'>
+                  <img src={item.imageURL}/>
+                </div>
+                <div className='item-desc'>
+                  <span className="title">{item.name}</span>
+                  <p><b>Price: {item.price}$</b></p>
+                  <p>
+                      <b>Quantity:</b>
+                  </p>
+                </div>
+                  <div className='add-remove'>
+                  <i className="material-icons">arrow_drop_up</i>
+                  <i className="material-icons">arrow_drop_down</i>
+                  </div>
+                <button className='waves-effect waves-light btn orange darken-3 remove'onClick={() => destroy(item)}>Remove Item</button>
+              </li>))
+          }
+          </ul>
+          <h3>Subtotal ${cartTotal}</h3>
+          <button>Submit Order</button>
+        </div>
+      </div>
+    );
+  }
+}
+
+const Cart = connect(({ cart, guestCart, products, orders, auth }) => {
+  return {
+    cart,
+    guestCart,
+    products,
+    orders,
+    auth
+  };
+}, (dispatch) => {
+  return {
+    destroy: item => dispatch(deleteCartItemThunk(item)),
+    deleteGuestItem: item => dispatch(deleteGuestItem(item))
+  }
+})(_Cart);
+export default Cart;
```

Comment:
Lot going on here... I think we could simplify a lot of this logic, by having smarter use of backend routes that give us the correct information, so that we don't have to do all this transformation on the FE.